packages/grid/backend/backend.dockerfile

```diff
@@ -1,94 +1,43 @@
-ARG PYTHON_VERSION="3.12"
-ARG TZ="Etc/UTC"
-
-# change to USER="syftuser", UID=1000 and HOME="/home/$USER" for rootless
-ARG USER="root"
-ARG UID=0
-ARG USER_GRP=$USER:$USER
-ARG HOME="/root"
-ARG APPDIR="$HOME/app"
-
-# ==================== [BUILD STEP] Python Dev Base ==================== #
-
-FROM cgr.dev/chainguard/wolfi-base as python_dev
-
-ARG PYTHON_VERSION
-ARG TZ
-ARG USER
-ARG UID
-
-# Setup Python DEV
-RUN --mount=type=cache,target=/var/cache/apk,sharing=locked \
-    apk update && \
-    apk upgrade && \
-    apk add build-base gcc tzdata python-$PYTHON_VERSION-dev-default py$PYTHON_VERSION-pip && \
-    ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-# uncomment for creating rootless user
-# && adduser -D -u $UID $USER
+FROM cgr.dev/chainguard/wolfi-base as backend
 
-# ==================== [BUILD STEP] Install Syft Dependency ==================== #
+ARG PYTHON_VERSION="3.12"
 
-FROM python_dev as syft_deps
+RUN apk update && apk upgrade && \
+    apk add git bash python-$PYTHON_VERSION-default uv=0.1.22-r0
 
-ARG APPDIR
-ARG HOME
-ARG UID
-ARG USER
-ARG USER_GRP
+WORKDIR /root/app
 
-USER $USER
-WORKDIR $APPDIR
-ENV PATH=$PATH:$HOME/.local/bin
+# keep static deps separate to have each layer cached independently
 
-# copy skeleton to do package install
-COPY --chown=$USER_GRP \
-    syft/setup.py \
-    syft/setup.cfg \
-    syft/pyproject.toml \
-    syft/MANIFEST.in \
-    syft/
+RUN --mount=type=cache,target=/root/.cache,sharing=locked \
+    uv venv && \
+    uv pip install torch==2.2.1+cpu --index-url https://download.pytorch.org/whl/cpu
 
-COPY --chown=$USER_GRP \
-    syft/src/syft/VERSION \
-    syft/src/syft/capnp \
-    syft/src/syft/
+RUN --mount=type=cache,target=/root/.cache,sharing=locked \
+    uv pip install jupyterlab==4.1.5
```

Comment:
From testing, jupyterlab or plain jupyter depends on psutil.

and Currently psutil does not support aarch64 wheels.

This would make it fail on mac silicon chips, if python3-dev is not installed.

The error message, which I received during building.
![Screenshot 2024-03-21 at 1 29 39 PM](https://github.com/OpenMined/PySyft/assets/43314053/48e42c80-3c13-4c88-b5a7-37f12049918b)

Reply:
Ah yeah i figured this was going to break - nevermind i have the python-dev way implemented as well, i'll just update the file to use it

Reply:
fixed